### PR TITLE
Fix successive calls of :meth:`.LinearTransformationScene.apply_matrix`

### LINKED ISSUE #2713: Cannot perform successive apply_matrix() in LinearTransformationScene
## Description of bug / unexpected behavior

4 months ago, I was able to run code to perform successive apply_matrix() in LinearTransformationScene, and created videos of them. I added self.wait(0) in between each.

However, after revisiting manim today, now it gives an error when performing the second apply_matrix(), even though the code is the same. It is able to perform one apply_matrix().

## Expected behavior

The code was able to create a video last time.

## How to reproduce the issue

<details><summary>Code for reproducing the problem</summary>

```py
from manim import *
class test(LinearTransformationScene):
    def __init__(self):
        LinearTransformationScene.__init__(
            self,
        )
    def construct(self):
        matrix = [[-0.3, 0.9], [0.7, 0.4]]
        self.apply_matrix(matrix)
        self.wait(0)
        
        matrix = [[1, -1], [1, 1]]
        self.apply_matrix(matrix)
        self.wait(0)

```

Or the code from this link's solutions:
https://stackoverflow.com/questions/69203723/how-to-apply-two-tranformations-one-after-the-other-in-manimce
</details>

## Logs
<details><summary>Terminal output</summary>

```
Manim Community v0.15.2

[05/09/22 00:05:39] INFO     Animation 0 : Using cached data (hash :                               cairo_renderer.py:75
                             2724823860_80563834_388161424)
                    INFO     Animation 1 : Using cached data (hash :                               cairo_renderer.py:75
                             1164547788_1764062066_2625276077)
┌─────────────────────────────── Traceback (most recent call last) ────────────────────────────────┐
│                                                                                                  │
│ C:\Users\a\Documents\github\.conda\envs\manim\lib\site-packages\manim\cli\render\comma │
│ nds.py:118 in render                                                                             │
│                                                                                                  │
│   115 │   │   for SceneClass in scene_classes_from_file(file):                                   │
│   116 │   │   │   try:                                                                           │
│   117 │   │   │   │   scene = SceneClass()                                                       │
│ > 118 │   │   │   │   scene.render()                                                             │
│   119 │   │   │   except Exception:                                                              │
│   120 │   │   │   │   error_console.print_exception()                                            │
│   121 │   │   │   │   sys.exit(1)                                                                │
│ C:\Users\a\Documents\github\.conda\envs\manim\lib\site-packages\manim\scene\scene.py:2 │
│ 22 in render                                                                                     │
│                                                                                                  │
│    219 │   │   """                                                                               │
│    220 │   │   self.setup()                                                                      │
│    221 │   │   try:                                                                              │
│ >  222 │   │   │   self.construct()                                                              │
│    223 │   │   except EndSceneEarlyException:                                                    │
│    224 │   │   │   pass                                                                          │
│    225 │   │   except RerunSceneException as e:                                                  │
│                                                                                                  │
│ D:\Documents\\manim\test.py:13 in construct                                │
│                                                                                                  │
│   10 │   │   self.wait(0)                                                                        │
│   11 │   │                                                                                       │
│   12 │   │   matrix = [[1, -1], [1, 1]]                                                          │
│ > 13 │   │   self.apply_matrix(matrix)                                                           │
│   14 │   │   self.wait(0)                                                                        │
│                                                                                                  │
│ C:\Users\a\Documents\github\.conda\envs\manim\lib\site-packages\manim\scene\vector_spa │
│ ce_scene.py:1042 in apply_matrix                                                                 │
│                                                                                                  │
│   1039 │   │   **kwargs                                                                          │
│   1040 │   │   │   Any valid keyword argument of self.apply_transposed_matrix()                  │
│   1041 │   │   """                                                                               │
│ > 1042 │   │   self.apply_transposed_matrix(np.array(matrix).T, **kwargs)                        │
│   1043 │                                                                                         │
│   1044 │   def apply_inverse(self, matrix, **kwargs):                                            │
│   1045 │   │   """                                                                               │
│                                                                                                  │
│ C:\Users\a\Documents\github\.conda\envs\manim\lib\site-packages\manim\scene\vector_spa │
│ ce_scene.py:1078 in apply_transposed_matrix                                                      │
│                                                                                                  │
│   1075 │   │   │   │   [angle_of_vector(func(RIGHT)), angle_of_vector(func(UP)) - np.pi / 2],    │
│   1076 │   │   │   )                                                                             │
│   1077 │   │   │   kwargs["path_arc"] = net_rotation                                             │
│ > 1078 │   │   self.apply_function(func, **kwargs)                                               │
│   1079 │                                                                                         │
│   1080 │   def apply_inverse_transpose(self, t_matrix, **kwargs):                                │
│   1081 │   │   """                                                                               │
│                                                                                                  │
│ C:\Users\a\Documents\github\.conda\envs\manim\lib\site-packages\manim\scene\vector_spa │
│ ce_scene.py:1141 in apply_function                                                               │
│                                                                                                  │
│   1138 │   │   │   + [                                                                           │
│   1139 │   │   │   │   self.get_vector_movement(function),                                       │
│   1140 │   │   │   │   self.get_transformable_label_movement(),                                  │
│ > 1141 │   │   │   │   self.get_moving_mobject_movement(function),                               │
│   1142 │   │   │   ]                                                                             │
│   1143 │   │   │   + [Animation(f_mob) for f_mob in self.foreground_mobjects]                    │
│   1144 │   │   │   + added_anims                                                                 │
│                                                                                                  │
│ C:\Users\a\Documents\github\.conda\envs\manim\lib\site-packages\manim\scene\vector_spa │
│ ce_scene.py:986 in get_moving_mobject_movement                                                   │
│                                                                                                  │
│    983 │   │   │   │   m.target = m.copy()                                                       │
│    984 │   │   │   target_point = func(m.get_center())                                           │
│    985 │   │   │   m.target.move_to(target_point)                                                │
│ >  986 │   │   return self.get_piece_movement(self.moving_mobjects)                              │
│    987 │                                                                                         │
│    988 │   def get_vector_movement(self, func):                                                  │
│    989 │   │   """                                                                               │
│                                                                                                  │
│ C:\Users\a\Documents\github\.conda\envs\manim\lib\site-packages\manim\scene\vector_spa │
│ ce_scene.py:957 in get_piece_movement                                                            │
│                                                                                                  │
│    954 │   │   Animation                                                                         │
│    955 │   │   │   The animation of the movement.                                                │
│    956 │   │   """                                                                               │
│ >  957 │   │   start = VGroup(*pieces)                                                           │
│    958 │   │   target = VGroup(*(mob.target for mob in pieces))                                  │
│    959 │   │   if self.leave_ghost_vectors:                                                      │
│    960 │   │   │   self.add(start.copy().fade(0.7))                                              │
│                                                                                                  │
│ C:\Users\a\Documents\github\.conda\envs\manim\lib\site-packages\manim\mobject\types\ve │
│ ctorized_mobject.py:1779 in __init__                                                             │
│                                                                                                  │
│   1776 │                                                                                         │
│   1777 │   def __init__(self, *vmobjects, **kwargs):                                             │
│   1778 │   │   super().__init__(**kwargs)                                                        │
│ > 1779 │   │   self.add(*vmobjects)                                                              │
│   1780 │                                                                                         │
│   1781 │   def __repr__(self):                                                                   │
│   1782 │   │   return (                                                                          │
│                                                                                                  │
│ C:\Users\a\Documents\github\.conda\envs\manim\lib\site-packages\manim\mobject\types\ve │
│ ctorized_mobject.py:1844 in add                                                                  │
│                                                                                                  │
│   1841 │   │   │   │   │   )                                                                     │
│   1842 │   │   """                                                                               │
│   1843 │   │   if not all(isinstance(m, (VMobject, OpenGLVMobject)) for m in vmobjects):         │
│ > 1844 │   │   │   raise TypeError("All submobjects must be of type VMobject")                   │
│   1845 │   │   return super().add(*vmobjects)                                                    │
│   1846 │                                                                                         │
│   1847 │   def __add__(self, vmobject):                                                          │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
TypeError: All submobjects must be of type VMobject
```

</details>

## System specifications

<details><summary>System Details</summary>

```

- OS- Windows 10
- RAM: 16GB
- Python version (`python/py/python3 --version`): 3.9.7
- Installed modules (provide output from `pip list`):
(too many installed modules to list, using manim 0.15.2)
```
</details>

### PR DESCRIPTION
**Pull request by SirJamesClarkMaxwell**
## Overview: What does this pull request change?

I have, probably solved this [issue][(https://github.com/ManimCommunity/manim/issues/3479](https://github.com/ManimCommunity/manim/issues/2713))
to reproduce the original problem:
```py
class Test(LinearTransformationScene):
    def construct(self):
        matrix = [[3, 0], [1, -1]]

        self.add_unit_square()
        c = Circle(radius=0.5)
        self.add_moving_mobject(c)
        self.wait()
        self.apply_matrix(matrix)
        self.wait()
        self.apply_inverse(matrix)
```
I have found two potential solutions:
1. filter in `self.get_piece_movement` that ` pieces`, which is in the original bug `self.moving_mobjects`, are VMobjects, and this new variable passes instead of the old one
2. This is not in this commit, but after I made this commit I recognized this possibility: change the from `VGroup` to `Group` 

In both solutions, the code provided above is working, and honestly, I don't know which solution would be better. 

### Why we encounter this bug 
Aside from the obvious reason that the author didn't check the type of objects of `pieces` passed into the `get_pieces_movement`, one of the reasons is that the Animation class requires some Mobject to work, even if this is an empty object or None, later in the code, completely not related in the main issue, there is a method that is restoring all objects in the scene, including this passed into the animation, these objects are injected into self.moving_mobjects -> and this information propagated into our function, `get_moving_mobject_movement`.

The really interesting thing is in the `self.add_mobjects_from_animations` method from the "scene" file. Because the Wait() animation has marked the None mobject parameter. Here is the main issue, in Animation class we could see this line
`self.mobject: Mobject = mobject if mobject is not None else Mobject()` and this is the reason why in `add_mobjects_from_animations` checking that object is not none isn't working. Because the Mobject is created for every animation object. 

## Links to added or changed documentation pages

closes #2713 

## Further Information and Comments

I don't know why the changes from the other branch were included in this commit ..., please ignore it.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: LinearTransformBug.py
Comment (JasonGrace2282) [inline LinearTransformBug.py:1]: Please remove this file from the PR. Something like `git rm LinearTransformBug.py` should work.

File: docs/source/faq/installation.md
Comment (JasonGrace2282) [inline docs/source/faq/installation.md:1]: You can get rid of this file from the commit history by looking at [this stack overflow post](https://stackoverflow.com/a/30893371/19916174)

File: manim/scene/vector_space_scene.py [hunk 1]
```diff
@@ -1000,17 +1000,21 @@ def get_piece_movement(self, pieces: list | tuple | np.ndarray):
 
         Returns
         -------
-        Animation
+        Animation`
```
Comment (JasonGrace2282) [inline manim/scene/vector_space_scene.py:1003]:
```suggestion
        Animation
```
Was this just a mistake?

File: manim/scene/vector_space_scene.py [hunk 2]
```diff
@@ -1000,17 +1000,21 @@ def get_piece_movement(self, pieces: list | tuple | np.ndarray):
...
-            self.ghost_vectors.add(start.copy().fade(0.7))
-            self.add(self.ghost_vectors[-1])
-        return Transform(start, target, lag_ratio=0)
+        if v_pieces:
+            if self.leave_ghost_vectors and start.submobjects:
+                # start.copy() gives a VGroup of Vectors
+                self.ghost_vectors.add(start.copy().fade(0.7))
+                self.add(self.ghost_vectors[-1])
+            return Transform(start, target, lag_ratio=0)
+        return []
```
Comment (JasonGrace2282) [inline manim/scene/vector_space_scene.py:1017]:
```suggestion
        if self.leave_ghost_vectors and start.submobjects:
            # start.copy() gives a VGroup of Vectors
            self.ghost_vectors.add(start.copy().fade(0.7))
            self.add(self.ghost_vectors[-1])
        return Transform(start, target, lag_ratio=0)
```
I don't think there is a need for this check? If `v_pieces=[]`, then `start` would have no submobjects, so the ghost vectors wouldn't get added (which I'm not sure is important, based on your PR description). Regardless, you still want this function to return an `Animation`, not an empty list.

**Review by JasonGrace2282 — CHANGES_REQUESTED** [review]
Hey, thanks for tackling this issue!
I remember `LinearTransformationScene` was what I used to create my first animations, so this brings back memories of circumventing this bug, so it's nice to see it fixed :)
I've left a couple of comments, and I think it's important to add a test for this behavior to make sure it works in the future.

Thanks for helping make Manim better :sparkles:

File: manim/scene/vector_space_scene.py (hunk 1 above)
Comment (SirJamesClarkMaxwell) [reply manim/scene/vector_space_scene.py:1003]: yes
i didn't notice it

File: manim/scene/vector_space_scene.py (hunk 2 above)
Comment (SirJamesClarkMaxwell) [reply manim/scene/vector_space_scene.py:1017]: I had this in the back of my mind, but I saw in a comment, which I, unfortunately, removed xd, that we shouldn't pass an empty list of objects. but okay.

File: manim/scene/vector_space_scene.py [hunk 3]
```diff
@@ -1003,8 +1003,20 @@ def get_piece_movement(self, pieces: list | tuple | np.ndarray):
...
+        # NOTE other possible solution:
+        """
+        start = Group(*pieces)
+        target = Group(*(mob.target for mob in pieces))
+        """
+        # NOTE JasonGrace said that there is another possible solution
+        """
+        - self.mobject = mobject if mobject is not None else Mobject()
+        + self.mobject = mobject if mobject is not None else VMobject()
+        """
```
Comment (JasonGrace2282) [inline manim/scene/vector_space_scene.py:1019]:
```suggestion
```
This should not be added to the source code

**Review by JasonGrace2282 — COMMENTED** [review]
You don't need to put your solutions into the source code :)

File: manim/scene/vector_space_scene.py (hunk 2 above)
Comment (JasonGrace2282) [reply manim/scene/vector_space_scene.py]: I actually wrote that comment (xD) and all it meant was the if statement will only create ghost vectors if `start` is not an empty `VGroup` (hence the `and start.submobjects` part of the if condition). In this case, if `v_pieces=[]` then `start` is an empty `VGroup` and ghost vectors will not be added.

**Review by JasonGrace2282 — APPROVED** [review]
LGTM!

**Pull request merged**